scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -107,6 +107,21 @@ abstract class ScanResultsStorage {
         }
     }
 
+    /**
+     * The result of the [add] operation.
+     */
+    data class AddResult(
```

Comment:
I wanted to emphasize that this is only used for the `add` operation, not the other functions. With  a `StorageResult` I would expect all storage operations return such a result object (which might be worth considering, because it's better than throwing exceptions or returning empty objects with only logging the issue). To me `AddResult` feels right, because it is the result of the `add` operation.

Reply:
Well, I still don't like `AddResult`; but it's not worth arguing about it.